api/routes/generate.js

```diff
@@ -10,35 +10,30 @@ var thunkify = require('thunkify');
 // CRX packaging module, instantiated with the `new` keyword
 var Extension = require('crx');
 
+var knownHosts = {
```

Comment:
Less code when modifying is not necessarily easier to maintain...
Code is easy to maintain when it successfully follows the SoC pattern, which in the current code it does not.
When you separate the logic that gets the icon and the logic that gets the title to different function (it sounds obvious because it is two different logics) it helps you to follow the above pattern.

It is true what you said about when you add another website that needs to override both icon and title you have to do those 3 steps, but it only makes sense... You want to override the title so you have to change the logic of the function that returns the title, same for the icon. You are messing up with two different logics only bacacue they both use the host name as part of it. 

As for the "duplicate code" it is not true. When you do multiple switch case on the same object it does not count as multiple code. If, for, while, switch-case and other like tham are only helpers for you to create logics, when you do the same logic more than once it is code duplication, which is not the case here.